common/beerocks/bwl/nl80211/ap_wlan_hal_nl80211.cpp

```diff
@@ -268,7 +309,261 @@ bool ap_wlan_hal_nl80211::update_vap_credentials(
+            } else {
+                LOG(ERROR) << "Autoconfiguration: " << vap << " invalid authentication type";
+                abort = true;
+                return;
+            }
+
+            LOG(DEBUG) << "Autoconfiguration for ssid: " << bss_it->ssid
+                       << " auth_type: " << auth_type << " encr_type: " << enc_type
+                       << " network_key: " << bss_it->network_key
+                       << " fronthaul=" << beerocks::string_utils::bool_str(bss_it->fronthaul)
+                       << " backhaul=" << beerocks::string_utils::bool_str(bss_it->backhaul);
+
+            conf.set_create_vap_value(vap, "wps_state", bss_it->fronthaul ? "2" : "");
+            conf.set_create_vap_value(vap, "wps_independent", "0");
+            conf.set_create_vap_value(vap, "sFourAddrMode", bss_it->backhaul ? "1" : "");
+            conf.set_create_vap_value(vap, "max_num_sta", bss_it->backhaul ? "1" : "");
+
+            // oddly enough, multi_ap_backhaul_wpa_passphrase has to be
+            // quoted, while wpa_passphrase does not...
+            if (bss_it->fronthaul && !backhaul_wps_ssid.empty()) {
+                conf.set_create_vap_value(vap, "multi_ap_backhaul_ssid",
+                                          "\"" + backhaul_wps_ssid + "\"");
+                conf.set_create_vap_value(vap, "multi_ap_backhaul_wpa_passphrase",
+                                          backhaul_wps_passphrase);
+            }
+
+            conf.set_create_vap_value(vap, "wpa", wpa);
+            conf.set_create_vap_value(vap, "okc", okc);
+            conf.set_create_vap_value(vap, "wpa_key_mgmt", wpa_key_mgmt);
+            conf.set_create_vap_value(vap, "wpa_pairwise", wpa_pairwise);
+            conf.set_create_vap_value(vap, "wpa_psk", wpa_psk);
+            conf.set_create_vap_value(vap, "ieee80211w", ieee80211w);
+            conf.set_create_vap_value(vap, "wpa_passphrase", wpa_passphrase);
+            conf.set_create_vap_value(vap, "disable_pmksa_caching", disable_pmksa_caching);
+            conf.set_create_vap_value(vap, "wpa_disable_eapol_key_retries",
+                                      wpa_disable_eapol_key_retries);
+
+            // finally enable the vap (remove any previously set start_disabled)
+            conf.set_create_vap_value(vap, "start_disabled", "");
+
+        } else {
+            // no more data in the input bss-conf list
+            // disable the rest of the vaps
+            conf.set_create_vap_value(vap, "start_disabled", 1);
+            conf.set_create_vap_value(vap, "ssid", "");
+        }
+    };
+
+    conf.for_all_ap_vaps(configure_func, bss_info_conf_list.begin(), bss_info_conf_list.end(),
+                         [](const std::string &) { return true; });
+
+    if (abort) {
+        return false;
+    }
+
+    if (!conf.store()) {
+        LOG(ERROR) << "Autoconfiguration: cannot save hostapd config!";
+        return false;
+    }
+
+    const std::string cmd("UPDATE ");
```

Comment:
```suggestion
    const std::string cmd("UPDATE");
```

Is "UPDATE" a valid hostapd command? Where can I find the best document describing available hostapd commands?

Reply:
The source code is the only documentation I know of.

Reply:
and I think the space is mandatory. Anyhow I tested it _with_ space so I'll leave it there even if not needed.